### ISSUE TITLE
Cannot use token property in SlasherClientOptions

### ISSUE THREAD

**Issue by Romejanic** [Romejanic/slasher#11]
**Describe the bug**
Attempting to set the bot token via the token property in the client options results in the token being unable to be loaded. This is caused because the token property is removed from the options object before it can be read by the token loading code.

```js
// called BEFORE the token is read
function filterOptions(options) {
    let finalOptions = options;
    if (!options.intents) {
        finalOptions.intents = [discord_js_1.Intents.FLAGS.GUILDS];
    }
    else if (Array.isArray(options.intents)) {
        finalOptions.intents = [...options.intents, discord_js_1.Intents.FLAGS.GUILDS];
    }
    else if (typeof options.intents === "number") {
        finalOptions.intents = options.intents & discord_js_1.Intents.FLAGS.GUILDS;
    }
    else if (typeof options.intents === "string") {
        // this probably shouldn't be used, but maybe the user
        // needs a specific intent other than GUILDS? just leave
        // it be
        finalOptions.intents = options.intents;
    }
    delete options.token; // <-- HERE token is erroneously deleted
    return finalOptions;
}
```

**Comment by Romejanic** [comment]
Probably no reason to delete the token property. The object is only passed to the underlying discord.js client constructor which will do nothing with it.

**Comment by Romejanic** [comment]
Fixed in release 0.4.1
